### PR TITLE
Add issues to the TODOs in S.T.Json source for better tracking and minor clean up

### PR DESCRIPTION
**Pull request by ahsonkhan**
- Remove unnecessary check in condition:
  - https://github.com/dotnet/corefx/pull/42574#discussion_r345971819
  - https://github.com/dotnet/corefx/pull/39541/files#diff-e2e1b5ba65b1a8b77a086eb93e77c2d1R327
- Add issues to the TODOs for better tracking and fix some that were easy and could be addressed quickly.
  - https://github.com/dotnet/runtime/issues/32348
  - https://github.com/dotnet/runtime/issues/32350
  - https://github.com/dotnet/runtime/issues/32351
  - https://github.com/dotnet/runtime/issues/32352
  - https://github.com/dotnet/runtime/issues/32354
  - https://github.com/dotnet/runtime/issues/32355
  - https://github.com/dotnet/runtime/issues/32356
  - https://github.com/dotnet/runtime/issues/32357
  - https://github.com/dotnet/runtime/issues/32358

- Update ActiveIssue and enable tests that are now fixed.
  - https://github.com/dotnet/runtime/issues/32359

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs [hunk 1]
```diff
@@ -125,5 +126,17 @@ private class Polymorphic
             [JsonPropertyName("p_3")]
             public object P3 => "";
         }
+
+        // https://github.com/dotnet/corefx/issues/40979
```
Comment (Jozkee) [inline src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs]: https://github.com/dotnet/runtime/issues/30814 (https://github.com/dotnet/corefx/issues/40979) is closed, can you remove the comment?

**Review by jozkee — APPROVED** [review]
Otherwise, looks good.

File: src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs (hunk 1 above)
Comment (Jozkee) [reply src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs]: and also rename the test to reflect the dotnet/runtime issue number.

Comment (Jozkee) [reply src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs]: And also rename the test to reflect the dotnet/runtime issue number.

Comment (ahsonkhan) [reply src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs]: The link just points to where the sample test came from. We have such links in a few places where the unit test was there to capture a regression or if it was inspired by a user provided scenario as part of the issue they filed.

Comment (safern) [reply src/libraries/System.Text.Json/tests/Serialization/Object.WriteTests.cs]: Yeah I’ll update my PR to update all urls using my app.

**Comment by ahsonkhan** [comment]
CI failure is unrelated: https://github.com/dotnet/runtime/issues/32367

**Pull request merged**